examples/tutorials/plot_tuto_mcar.py

```diff
@@ -149,17 +163,221 @@
 # %%
-# The p-value is larger than 0.05, therefore we don't reject the HO MCAR assumption. In this case
-# this is a false negative since the missingness mechanism is MAR.
+# The Little's p-value is larger than 0.05, therefore, using this test we don't reject the H0 MCAR
+# assumption. In this case this is a false negative since the missingness mechanism is MAR.
+#
+# However the PKLM test p-value is smaller than 0.05 therefore we don't reject the H0 MCAR
+# assumption. In this case this is a true negative.
 
 # %%
-# Limitations
-# -----------
+# Limitations and conclusion
+# ==========================
 # In this tutoriel, we can see that Little's test fails to detect covariance heterogeneity between
 # patterns.
 #
 # We also note that the Little's test does not handle categorical data or temporally
 # correlated data.
+# 
+# This is why we have implemented the PKLM test, which makes up for the shortcomings of the Little
+# test. We present this test in more detail in the next section.
+
+# %%
+# 2. The PKLM test
+# ------------------------------------------------------------------
+#
+# The PKLM test is very powerful for several reasons. Firstly, it covers the concerns that Little's
+# test may have (covariance heterogeneity). Secondly, it is currently the only MCAR test applicable
+# to mixed data. Finally, it proposes a concept of partial p-value which enables us to carry out a
+# variable-by-variable diagnosis to identify the potential causes of a MAR mechanism.
+#
+# There is a parameter in the paper called size.res.set. The authors of the paper recommend setting
+# this parameter to 2. We have chosen to follow this advice and not leave the possibility of
+# increasing this parameter. The results are satisfactory and the code is simpler.
+#
+# It does have one disadvantage, however: its calculation time.
+#
+
+# %%
+
+"""
+Calculation time
+================
+
++------------+------------+----------------------+
+| **n_rows** | **n_cols** | **Calculation_time**  |
++============+============+======================+
+| 200        | 2          | 2"12                 |
++------------+------------+----------------------+
+| 500        | 2          | 2"24                 |
++------------+------------+----------------------+
+| 500        | 4          | 2"18                 |
++------------+------------+----------------------+
+| 1000       | 4          | 2"48                 |
++------------+------------+----------------------+
+| 1000       | 6          | 2"42                 |
++------------+------------+----------------------+
+| 10000      | 6          | 20"54                |
++------------+------------+----------------------+
+| 10000      | 10         | 14"48                |
++------------+------------+----------------------+
+| 100000     | 10         | 4'51"                |
++------------+------------+----------------------+
+| 100000     | 15         | 3'06"                |
++------------+------------+----------------------+
+"""
```

Comment:
Why, for the same number of lines, if the number of columns increases, the time decreases (the difference is more noticeable the more lines we have) ?

Reply:
I think that's because the .fit() of the classifier takes less time when there are more features.